react/features/recording/middleware.js

```diff
@@ -76,25 +84,48 @@ MiddlewareRegistry.register(store => next => action => {
         break;
     }
 
+    case HIDE_NOTIFICATION: {
+        const { pendingNotificationUid } = getState()['features/recording'];
+
+        if (action.uid === pendingNotificationUid) {
+            dispatch(clearPendingRecordingNotificationUid());
+        }
+        break;
+    }
+
     case RECORDING_SESSION_UPDATED: {
         const updatedSessionData
-            = getSessionById(store.getState(), action.sessionData.id);
+            = getSessionById(getState(), action.sessionData.id);
 
         if (updatedSessionData.mode === JitsiRecordingConstants.mode.FILE) {
-            const { OFF, ON } = JitsiRecordingConstants.status;
+            const { PENDING, OFF, ON } = JitsiRecordingConstants.status;
 
             if (updatedSessionData.status === ON
                 && (!oldSessionData || oldSessionData.status !== ON)) {
-                store.dispatch(playSound(RECORDING_ON_SOUND_ID));
+                dispatch(playSound(RECORDING_ON_SOUND_ID));
+                dispatch(hidePendingRecordingNotification());
             } else if (updatedSessionData.status === OFF
                 && (!oldSessionData || oldSessionData.status !== OFF)) {
-                store.dispatch(stopSound(RECORDING_ON_SOUND_ID));
-                store.dispatch(playSound(RECORDING_OFF_SOUND_ID));
+                dispatch(stopSound(RECORDING_ON_SOUND_ID));
+                dispatch(playSound(RECORDING_OFF_SOUND_ID));
+                dispatch(showStoppedRecordingNotification());
+            } else if (updatedSessionData.status === PENDING
+                && (!oldSessionData || oldSessionData.status !== PENDING)) {
+                dispatch(showPendingRecordingNotification());
             }
         }
 
         break;
     }
+
+    case SHOW_NOTIFICATION:
+        if (action.props.descriptionKey === 'recording.pending') {
```

Comment:
I wonder if it'll be better to sniff this out some other way, maybe like with an explicit flag. In case someone changes the string that displays and misses this reference.

Reply:
Okay, I check this, thanks!